src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -1,54 +1,150 @@
 package com.sprint.mission.discodeit.service.basic;
 
+import com.sprint.mission.discodeit.dto.channel.ChannelCreatePrivateDto;
+import com.sprint.mission.discodeit.dto.channel.ChannelCreatePublicDto;
+import com.sprint.mission.discodeit.dto.channel.ChannelResponseDto;
+import com.sprint.mission.discodeit.dto.channel.ChannelUpdateDto;
+import com.sprint.mission.discodeit.dto.readStatus.ReadStatusCreateDto;
 import com.sprint.mission.discodeit.entity.Channel;
 import com.sprint.mission.discodeit.entity.ChannelType;
+import com.sprint.mission.discodeit.entity.Message;
+import com.sprint.mission.discodeit.entity.ReadStatus;
 import com.sprint.mission.discodeit.repository.ChannelRepository;
+import com.sprint.mission.discodeit.repository.MessageRepository;
 import com.sprint.mission.discodeit.service.ChannelService;
+import com.sprint.mission.discodeit.service.ReadStatusService;
+import java.time.Instant;
+import java.util.Comparator;
 import java.util.List;
+import java.util.NoSuchElementException;
 import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
 
+@Service
+@RequiredArgsConstructor
 public class BasicChannelService implements ChannelService {
 
     private final ChannelRepository channelRepository;
+    private final ReadStatusService readStatusService;
+    private final MessageRepository messageRepository;
 
-    public BasicChannelService(ChannelRepository channelRepository) {
-        this.channelRepository = channelRepository;
+    @Override
+    public Channel createPrivate(ChannelCreatePrivateDto channelCreatePrivateDto) {
+
+        Channel newChannel = new Channel(ChannelType.PRIVATE, null, null);
```

Comment:
Channel 을 만들때 PRIVATE 혹은 PUBLIC 두가지의 Channel 을 만들게 되는데요
이때 생성자를 사용하기보다는 
정적 팩토리 메소드를 활용하면 더 명확하게 Channel 을 생성할 수 있을것 같아요 
(참고:
https://inpa.tistory.com/entry/GOF-%F0%9F%92%A0-%EC%A0%95%EC%A0%81-%ED%8C%A9%ED%86%A0%EB%A6%AC-%EB%A9%94%EC%84%9C%EB%93%9C-%EC%83%9D%EC%84%B1%EC%9E%90-%EB%8C%80%EC%8B%A0-%EC%82%AC%EC%9A%A9%ED%95%98%EC%9E%90)


예를들어 Channel 클래스안에

```suggestion
      public static Channel createPrivate(){
          return new Channel(ChannelType.PRIVATE, null, null);
      }
      
      public static Channel createPublic(String channelName, String description){
       return new Channel(ChannelType.PUBLIC, channelName,
                description);
      }
    
```
그리고 서비스에서는 위 두가지를 각각 호출하게 하는 방식으로 하면 Private 인 경우 나머지가 null 이 되어야한다는 무결성을 실수없이 넣을 수 있고, 퍼블릭인 경우는 name과 description 이 확실히 들어간다는 점도 알 수 있겠죠!